core/types/header_ext.go

```diff
@@ -86,7 +86,21 @@ func (h *HeaderExtra) DecodeJSON(eth *ethtypes.Header, input []byte) error {
 }
 
 func (h *HeaderExtra) PostCopy(dst *ethtypes.Header) {
-	panic("not implemented")
+	extraCopy := &HeaderExtra{
+		ExtDataHash: h.ExtDataHash,
+	}
+
+	if h.BlockGasCost != nil {
+		extraCopy.BlockGasCost = big.NewInt(0)
+		extraCopy.BlockGasCost.SetBytes(h.BlockGasCost.Bytes())
+	}
+
+	if h.ExtDataGasUsed != nil {
+		extraCopy.ExtDataGasUsed = big.NewInt(0)
+		extraCopy.ExtDataGasUsed.SetBytes(h.ExtDataGasUsed.Bytes())
+	}
+
+	_ = WithHeaderExtra(dst, extraCopy)
```

Comment:
Probably beyond the scope of this PR, but this pattern suggests that `WithHeaderExtra` is incorrectly named as it should be `SetHeaderExtra`. The `With` prefix implies that it's returning a value but not _necessarily_ that it's setting it on the argument.

Reply:
This comment was only meant to be on the final line.

Reply:
Indeed changed in base branch from WithHeaderExtra -> SetHeaderExtra  0ecd53e973f12591adc0e1007f535898f285ae7d